src/general/StageBase.cs

```diff
@@ -285,6 +285,9 @@ public virtual void OnReturnFromEditor()
         if (CurrentGame == null)
             throw new InvalidOperationException("Returning to stage from editor without a game setup");
 
+        // Update the generation history with the newly edited player species
+        GameWorld.GenerationHistory.Last().Value.UpdateSpeciesData(GameWorld.PlayerSpecies);
```

Comment:
This relies on the dictionary being in-order, which is not 100% guaranteed, so the history should actually be a list or what I'll do myself to fix this right now is to find the max key and use that here.